Apps/APN-UIKit/APN UIKit/APN UIKit/View/SettingsViewController.swift

```diff
@@ -18,23 +18,56 @@ class SettingsViewController: UIViewController {
     @IBOutlet weak var bgQMinTasksTextField: ThemeTextField!
     @IBOutlet weak var bgQTakDelayTextField: ThemeTextField!
     
+    var notificationUtil = DI.shared.notificationUtil
     var settingsRouter: SettingsRouting?
     var storage = DI.shared.storage
     var currentSettings : Settings!
+    
+    var pushSwitchState:Bool {
+        return enablePushToggle.isOn
+    }
+    
+    var trackScreenState:Bool {
+        return trackScreenToggle.isOn
+    }
+    
+    var trackDeviceAttributeState:Bool {
+        return trackDeviceToggle.isOn
+    }
+    
+    var debugModeState: Bool {
+        return debugModeToggle.isOn
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
         configureSettingsRouter()
-        getDefaultValues()
-        setDefaultValues()
+        getAndSetDefaultValues()
+        addObserversForSettingsScreen()
     }
     
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         self.navigationController?.isNavigationBarHidden = false
     }
     
-    func getDefaultValues() {
+    func addObserversForSettingsScreen() {
+        let nc = NotificationCenter.default
+        nc.addObserver(self, selector: #selector(appMovedToForeground), name: UIApplication.willEnterForegroundNotification, object: nil)
```

Comment:
I am assuming that using an observer for push state is because someone could go into the iOS Settings app and change the push permissions for this app and then re-open this app? 

If that assumption is correct, I don't think we need an observer for that. I think that when settings are changed for an app in the Settings iOS app, the app is force killed by the OS. Because the app would be killed and re-opened, this observer wouldn't do anything.

Reply:
If we do indeed need this observer, don't forget to remove the observer in `deinit`